src/main/java/dev/waterdog/utils/Configuration.java

```diff
@@ -59,91 +83,147 @@ public void loadFrom(Map<String, Object> values) {
     }
 
     public Set<String> getKeys() {
-        return new HashSet<>(this.values.keySet());
+        return this.values.keySet();
+    }
+
+    public void remove(String key) {
+        LastMap lastMap = getLastMap(key);
```

Comment:
Please try to use same code style. If  method or value refers to current object, please use `this` when calling value/method. Please go thought all changes and correct it.